core/src/main/java/ai/timefold/solver/core/impl/solver/termination/UnimprovedBestSolutionTermination.java

```diff
@@ -0,0 +1,178 @@
+package ai.timefold.solver.core.impl.solver.termination;
+
+import ai.timefold.solver.core.api.score.Score;
+import ai.timefold.solver.core.impl.phase.scope.AbstractPhaseScope;
+import ai.timefold.solver.core.impl.phase.scope.AbstractStepScope;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+import ai.timefold.solver.core.impl.solver.thread.ChildThreadType;
+
+public final class UnimprovedBestSolutionTermination<Solution_> extends AbstractTermination<Solution_> {
+
+    // Minimal interval of time to avoid early conclusions
+    protected static final long MINIMAL_INTERVAL_TIME = 10L;
+    // The ratio specifies the minimum criteria to determine a flat line between two move count values.
+    // A value of 0.2 represents 20% of the execution time of the current growth curve.
+    // For example, the first best solution is found at 0 seconds,
+    // while the last best solution is found at 60 seconds.
+    // Given the total time of 60 seconds,
+    // we will identify a flat line between the last best solution and the discovered new best solution
+    // if the time difference exceeds 12 seconds.
+    private final double maxUnimprovedBestSolutionLimit;
```

Comment:
Needs a better name, describing the functionality.
This also needs to show up in the docs and in the images, as it is one of the key parameters to understand the algorithm.